google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -395,22 +425,208 @@ public void testGetOptions() {
 
   @Test
   public void testCreateBucket() {
-    when(storageRpcMock.create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS))
-        .thenReturn(BUCKET_INFO1.toPb())
-        .thenThrow(new RuntimeException("Fail"));
+    doReturn(BUCKET_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS);
     initializeService();
     Bucket bucket = storage.create(BUCKET_INFO1);
     assertEquals(expectedBucket1, bucket);
   }
 
   @Test
   public void testCreateBucketWithOptions() {
-    when(storageRpcMock.create(BUCKET_INFO1.toPb(), BUCKET_TARGET_OPTIONS))
-        .thenReturn(BUCKET_INFO1.toPb())
-        .thenThrow(new RuntimeException("Fail"));
+    doReturn(BUCKET_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .create(BUCKET_INFO1.toPb(), BUCKET_TARGET_OPTIONS);
     initializeService();
     Bucket bucket =
         storage.create(BUCKET_INFO1, BUCKET_TARGET_METAGENERATION, BUCKET_TARGET_PREDEFINED_ACL);
     assertEquals(expectedBucket1, bucket);
   }
+
+  @Test
+  public void testCreateBucketFailure() {
+    doThrow(STORAGE_FAILURE).when(storageRpcMock).create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS);
+    initializeService();
+    try {
+      storage.create(BUCKET_INFO1);
+      fail();
+    } catch (StorageException e) {
+      assertEquals(STORAGE_FAILURE, e.getCause());
+    }
+  }
+
+  private void verifyChannelRead(ReadChannel channel, byte[] bytes) throws IOException {
+    assertNotNull(channel);
+    assertTrue(channel.isOpen());
+
+    ByteBuffer buffer = ByteBuffer.allocate(42);
+    byte[] expectedBytes = new byte[buffer.capacity()];
+    System.arraycopy(bytes, 0, expectedBytes, 0, bytes.length);
+
+    int size = channel.read(buffer);
+    assertEquals(bytes.length, size);
+    assertEquals(bytes.length, buffer.position());
+    assertArrayEquals(expectedBytes, buffer.array());
+  }
+
+  @Test
+  public void testReader() {
+    initializeService();
+    ReadChannel channel = storage.reader(BUCKET_NAME1, BLOB_NAME1);
+    assertNotNull(channel);
+    assertTrue(channel.isOpen());
+    // Storage.reader() does not issue any RPC, channel.read() does
+    try {
+      channel.read(ByteBuffer.allocate(100));
```

Comment:
Can we please add `fail()` after this line.

Reply:
Thanks for catching that!